src/components/profile/edit-profile.js

```diff
@@ -1,113 +1,235 @@
-import React from "react";
-import {Link} from "react-router-dom";
-
+/**
+ * @file Implements edit-profile component that displays user's information, and
+ * enable them to manage their account.
+ * Users can click update to update their profile information
+ * Users can click delete to delete their account
+ * Users can click go-back to their profile page.
+ */
+ import React, {useEffect, useState} from "react";
+ import {Link, useNavigate} from "react-router-dom";
+ import * as authService from "../../services/auth-service";
+ import * as userService from "../../services/users-service";
+ 
+ /**
+  * Implements EditProfile component that will fetch and display user's profile information and
+  * enables users to edit their profile information and delete their account
+  */
 const EditProfile = () => {
+  const navigate = useNavigate();
+    const [profileInfo, setProfileInfo] = useState({});
+    const [passwordChanged, setPasswordChanged] = useState(false);
+
+    useEffect(() => {
+        authService.profile()
+            .then(profile => {
+                console.log(profile)
+                setProfileInfo(profile);
+            })
+    }, [])
+
+    /**
+     * A call back function that helps call user service
+     * to update user's profile information when user
+     * clicks the "save" button
+     */
+    const handleUserUpdate = () => {
+        if (!passwordChanged) {
+            delete profileInfo.password
+        }
+        let newProfile = profileInfo;
+        if (profileInfo.password === '') {
+            alert("Password cannot be empty")
+        } else {
+            console.log("update to it", newProfile)
```

Comment:
Probably don't want the console.logs kept in there if its exposing the underlying data for our "production" code

Reply:
will change that

Reply:
Once that's changed, this merge is good to go in my book